### PR TITLE
Add integration tests for Spring Boot application example

### PR DESCRIPTION
**Pull request by lykims**
### :pencil: Description
Add integration tests for Spring Boot application example

### :link: Related Issues
Issue https://github.com/ExpediaGroup/graphql-kotlin/issues/389

### CODE REVIEW

File: detekt.yml [hunk 1]
```diff
@@ -2,21 +2,17 @@ build:
   weights:
     comments: 0   # missing comments should not fail the build
 
-test-pattern: # Configure exclusions for test sources
```
Comment (lykims) [inline detekt.yml]: `test-pattern` is not supported anymore with the latest version of Detekt.
https://arturbosch.github.io/detekt/configurations.html#path-filters--excludes--includes

File: examples/spring/pom.xml [hunk 2]
```diff
@@ -91,6 +105,25 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>build-helper-maven-plugin</artifactId>
```
Comment (dariuszkuc) [inline examples/spring/pom.xml]: there is no need for this plugin, lets move the IT classes under `test` directory

File: examples/spring/src/integration-test/kotlin/com/expediagroup/graphql/examples/Constants.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.examples
+
+import org.springframework.http.MediaType
+
+object Constants {
```
Comment (dariuszkuc) [inline examples/spring/src/integration-test/kotlin/com/expediagroup/graphql/examples/Constants.kt]: no need to wrap those constants in an object -> lets rename the class to `IntegrationTestConstants` and have those const values as top level declarations

File: examples/spring/src/integration-test/kotlin/com/expediagroup/graphql/examples/IntegrationTest.kt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+package com.expediagroup.graphql.examples
+
+import com.expediagroup.graphql.examples.Constants.DATA_JSON_PATH
+import com.expediagroup.graphql.examples.Constants.ERRORS_JSON_PATH
+import com.expediagroup.graphql.examples.Constants.EXTENSIONS_JSON_PATH
+import org.springframework.test.web.reactive.server.WebTestClient
+
+interface IntegrationTest {
```
Comment (dariuszkuc) [inline examples/spring/src/integration-test/kotlin/com/expediagroup/graphql/examples/IntegrationTest.kt]: there is no need to define this interface, lets rename this file as `WebTestClientExtensions` and move those to top level functions instead

File: examples/spring/pom.xml (hunk 2 above)
Comment (lykims) [reply examples/spring/pom.xml]: Hi Dariusz,
I moved the IT classes under `test` directory.

File: examples/spring/src/integration-test/kotlin/com/expediagroup/graphql/examples/Constants.kt (hunk 3 above)
Comment (lykims) [reply examples/spring/src/integration-test/kotlin/com/expediagroup/graphql/examples/Constants.kt]: I renamed the class, made the constants as top level declarations, and moved it under `test` directory.

File: examples/spring/src/integration-test/kotlin/com/expediagroup/graphql/examples/IntegrationTest.kt (hunk 4 above)
Comment (lykims) [reply examples/spring/src/integration-test/kotlin/com/expediagroup/graphql/examples/IntegrationTest.kt]: I renamed the class, made the functions as top level, and moved the class under `test` directory.

**Pull request merged**